gallery_dl/extractor/fanbox.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.fanbox.cc/"""
+
+from .common import Extractor, Message
+from .. import text
+
+
+BASE_PATTERN = (
+    r"(?:https?://)?(?:([a-zA-Z0-9_-]+)\.fanbox\.cc|"
+    r"(?:www\.)?fanbox\.cc/@([^/?#]+))"
+)
+
+
+class FanboxExtractor(Extractor):
+    """Base class for Fanbox extractors"""
+    category = "fanbox"
+    root = "https://www.fanbox.cc"
+    directory_fmt = ("{category}", "{creatorId}")
+    filename_fmt = "{id}_{num}.{extension}"
+    archive_fmt = "{id}_{num}"
+    _warning = True
+
+    def items(self):
+        yield Message.Version, 1
+
+        if self._warning:
+            if "FANBOXSESSID" not in self.session.cookies:
+                self.log.warning("no 'FANBOXSESSID' cookie set")
+            FanboxExtractor._warning = False
+
+        for content_body, post in self.posts():
+            yield Message.Directory, post
+            yield from self._get_urls_from_post(content_body, post)
+
+    def posts(self):
+        """Return all relevant post objects"""
+
+    def _pagination(self, url):
+        headers = {"Origin": self.root}
+
+        while url:
+            url = text.ensure_http_scheme(url)
+            body = self.request(url, headers=headers).json()["body"]
+            for item in body["items"]:
+                yield self._get_post_data(item["id"])
+
+            url = body["nextUrl"]
```

Comment:
The items returned from `https://api.fanbox.cc/post.listCreator?creatorId=USER&limit=10` appear to be, at least for `xub.fanbox.cc`, more or less the same as the single-item results from `https://api.fanbox.cc/post.info?postId=ID` (*) It's only missing comments, `imageForShare`, and the entries about next/previous posts.

If that's true in general (posts with videos, embeds, etc), we don't need to fetch data from `/post.info` for every post and can use the items returned from `/post.listCreator` directly.

<details>
 <summary>(*) Diff "/post.listCreator" - "/post.info" for post 2059366</summary>

``` diff
   "creatorId": "xub",
-  "hasAdultContent": true,
-  "commentList": {
-    "items": [],
-    "nextUrl": null
-  },
-  "nextPost": {
-    "id": "2085876",
-    "title": "Skeb Commission",
-    "publishedDatetime": "2021-04-03 04:56:12"
-  },
-  "prevPost": {
-    "id": "2009099",
-    "title": "ﾒｽｶﾞｷ〇〇★〇〇〇〇〇をわからせる絵",
-    "publishedDatetime": "2021-03-12 19:01:08"
-  },
-  "imageForShare": "https://pixiv.pximg.net/c/1200x630_90_a2_g5/fanbox/public/images/post/2059366/cover/JuTXNtvo1BRN93cLW371vVd6.jpeg"
+  "hasAdultContent": true
 }

```
</details>

Reply:
Seems like you are right, at least I couldn't find any posts where listCreator didn't have all the content. Updated the code to use that directly, it's easy enough to change it back if it turns out it's needed.